Framework/Algorithms/src/SofQWNormalisedPolygon.cpp

```diff
@@ -113,23 +112,19 @@ void SofQWNormalisedPolygon::exec() {
     if (spectrumInfo.isMasked(i) || spectrumInfo.isMonitor(i)) {
       continue;
     }
+    const auto &det = spectrumInfo.detector(i);
 
-    double theta = this->m_theta[i];
-    double phi = this->m_phi[i];
-    double thetaWidth = this->m_thetaWidths[i];
-    double phiWidth = this->m_phiWidths[i];
+    const double theta = this->m_theta[i];
+    const double phi = this->m_phi[i];
+    const double thetaWidth = this->m_thetaWidths[i];
+    const double phiWidth = this->m_phiWidths[i];
 
     // Compute polygon points
-    double thetaHalfWidth = 0.5 * thetaWidth;
-    double phiHalfWidth = 0.5 * phiWidth;
+    const double thetaHalfWidth = 0.5 * thetaWidth;
 
     const double thetaLower = theta - thetaHalfWidth;
     const double thetaUpper = theta + thetaHalfWidth;
 
-    const double phiLower = phi - phiHalfWidth;
-    const double phiUpper = phi + phiHalfWidth;
-
-    const double efixed = m_EmodeProperties.getEFixed(spectrumInfo.detector(i));
```

Comment:
Hmm, this is also using `getEFixed` for a (potential) `DetectorGroup`, does this imply that it does the right thing...?

Edit: tried to track it down: I believe `getEFixed` simply throw in that case?

Reply:
`DetectorGroup::getNumberParameter()` always returns an empty `vector` so `SofQCommon::getEFixed()` would indeed throw for `DetectorGroup`.